samples/nrf9160/aws_fota/src/main.c

```diff
@@ -0,0 +1,584 @@
+		if (err) {
+			printk("NRF_CLOUD_CA_CERTIFICATE err: %d", err);
+			return err;
+		}
+
+		/* Provision Private Certificate. */
+		err = nrf_inbuilt_key_write(
+			NRF_CLOUD_SEC_TAG,
+			NRF_KEY_MGMT_CRED_TYPE_PRIVATE_CERT,
+			private_pem_key,
+			private_pem_key_len);
+		if (err) {
+			printk("NRF_CLOUD_CLIENT_PRIVATE_KEY err: %d", err);
+			return err;
+		}
+
+		/* Provision Public Certificate. */
+		err = nrf_inbuilt_key_write(
+			NRF_CLOUD_SEC_TAG,
+			NRF_KEY_MGMT_CRED_TYPE_PUBLIC_CERT,
+				 public_pem,
+				 public_pem_len);
+		if (err) {
+			printk("NRF_CLOUD_CLIENT_PUBLIC_CERTIFICATE err: %d",
+				err);
+			return err;
+		}
+	}
+#else
+  #error "Missing CONFIG_BSD_LIB"
+#endif
+}
+
+/* Function to get the client id */
+static int client_id_get(char *id)
+{
+#if !defined(NRF_CLOUD_CLIENT_ID)
+#if defined(CONFIG_BSD_LIBRARY)
+	int at_socket_fd;
+	int bytes_written;
+	int bytes_read;
+	char imei_buf[IMEI_LEN + 1];
+	int ret;
+
+	at_socket_fd = nrf_socket(NRF_AF_LTE, 0, NRF_PROTO_AT);
+	__ASSERT_NO_MSG(at_socket_fd >= 0);
+
+	bytes_written = nrf_write(at_socket_fd, "AT+CGSN", 7);
+	__ASSERT_NO_MSG(bytes_written == 7);
+
+	bytes_read = nrf_read(at_socket_fd, imei_buf, IMEI_LEN);
+	__ASSERT_NO_MSG(bytes_read == IMEI_LEN);
+	imei_buf[IMEI_LEN] = 0;
+
+	snprintf(id, CLIENT_ID_LEN + 1, "nrf-%s", imei_buf);
+
+	ret = nrf_close(at_socket_fd);
+	__ASSERT_NO_MSG(ret == 0);
+#else
+#error Missing NRF_CLOUD_CLIENT_ID
+#endif /* defined(CONFIG_BSD_LIBRARY) */
+#else
+	memcpy(id, CONFIG_CLOUD_CLIENT_ID, CLOUD_CLIENT_ID_LEN + 1);
+#endif /* !defined(NRF_CLOUD_CLIENT_ID) */
+
+	return 0;
+}
+
+/**@brief Initialize the MQTT client structure */
+static int client_init(struct mqtt_client *client)
+{
+	mqtt_client_init(client);
+	provision();
+	broker_init();
+
+	int ret = client_id_get(client_id_buf);
+
+	if (ret != 0) {
+		return ret;
+	}
+
+	/* MQTT client configuration */
+	client->broker = &broker;
+	client->evt_cb = mqtt_evt_handler;
+	client->client_id.utf8 = client_id_buf;
+	client->client_id.size = CLIENT_ID_LEN;
+	client->password = NULL;
+	client->user_name = NULL;
+	client->protocol_version = MQTT_VERSION_3_1_1;
+
+	/* MQTT buffers configuration */
+	client->rx_buf = rx_buffer;
+	client->rx_buf_size = sizeof(rx_buffer);
+	client->tx_buf = tx_buffer;
+	client->tx_buf_size = sizeof(tx_buffer);
+
+	/* MQTT transport configuration */
+	client->transport.type = MQTT_TRANSPORT_SECURE;
+
+	static sec_tag_t sec_tag_list[] = {NRF_CLOUD_SEC_TAG};
+	struct mqtt_sec_config *tls_config = &(client->transport).tls.config;
+
+	tls_config->peer_verify = 2;
+	tls_config->cipher_list = NULL;
+	tls_config->cipher_count = 0;
+	tls_config->sec_tag_count = ARRAY_SIZE(sec_tag_list);
+	tls_config->sec_tag_list = sec_tag_list;
+	tls_config->hostname = AWS_HOSTNAME;
+
+	return 0;
+
+}
+
+/**@brief Initialize the file descriptor structure used by poll. */
+static int fds_init(struct mqtt_client *c)
+{
+	if (c->transport.type == MQTT_TRANSPORT_NON_SECURE) {
```

Comment:
I think we only use `MQTT_TRANSPORT_SECURE` in this sample, I'd remove unnecessary code and leave `fds.fd = c->transport.tls.sock;`

Reply:
Fixed